webui/src/pages/repositories/repository/objects.jsx

```diff
@@ -137,7 +132,13 @@ const ImportButton = ({ config, repo, reference, path, onDone, onClick, variant
                 importBranchResp = await branches.get(repo.id, importBranch)
             } catch (error) {
                 if (error instanceof NotFoundError) {
-                    await branches.create(repo.id, importBranch, currBranch)
+                    // Find root commit for repository
+                    let response = await commits.log(repo.id, repo.default_branch, "", 1000)
+                    while (response.pagination.has_more) {
+                        response = await commits.log(repo.id, repo.default_branch, response.pagination.next_offset, 1000)
+                    }
```

Comment:
Does JS has a `do {...} while(...)` option?

Reply:
There is a do-while but since I need the response I need to define it outside the scope of the while, I can separate the definition from the assignment if you think it looks better.
Can you recommend on a native JS reader to review? :D

Reply:
So maybe:

```suggestion
                    // Find root commit for repository
                    let hasMore = true;
                    let nextOffset = "";
                    let baseCommit = reference.id;
                    do {
                    	let response = await commits.log(repo.id, reference.id, nextOffset, 1000);
                    	hasMore = response.pagination.has_more;
                    	nextOffset = response.pagination.next_offset;
                    	baseCommit = response.results.at(-1);
                    } while (hasMore)
```